sdk/src/invariant.ts

```diff
@@ -528,6 +530,37 @@ export class Invariant {
       refTime: this.gasLimit.refTime.toNumber(),
       proofSize: this.gasLimit.proofSize.toNumber()
     }
+  ): Promise<[Position, Pool, Tick, Tick][]> {
+    const [positions, poolKeysCount] = await this.getPositions(
```

Comment:
IMO, there should be an option to skip the first fetch and consequently pass positionCount. This way, we can decrease the time and avoid the sequence of two fetches. 

At the moment of typing something in the search, you know how many positions there are and you probably also have one of the pages (at least).